web/client/plugins/widgets/WidgetsTray.jsx

```diff
@@ -126,6 +134,11 @@ export default compose(
         hasCollapsedWidgets: widgets.filter(({ collapsed } = {}) => collapsed).length > 0,
         hasTrayWidgets: widgets.length > 0
     })),
+    withProps(({ isMobileAgent }) => {
+        return {
+            isSingleWidgetLayout: isMobileAgent || window.innerWidth <= singleWidgetLayoutBreakpoint
+        };
```

Comment:
this should have a similar logic we applied in the Widgets plugin, see the resize.patch. We should avoid to use  window.innerWidth and use a width detector (eg the HOC `import { withResizeDetector } from 'react-resize-detector';` )

Reply:
Unfortunately react-resize-detector didn't work for WidgetsTray (undefined width and height), while react-container-dimensions as a HOC/enhancer worked well.